### DISCUSSION TITLE
Downloading tiles to SQLDelight and showing in map

### DISCUSSION THREAD

**Discussion by JakkeJ** [peterLaurence/MapCompose#109] [Q&A] [answered]
I am using this, but I need to get the tiles stored locally in case of loss of signal, and that part works, but it won't render tiles that aren't immediately available in the database, how could I get around this?

OSM>MapCompose works
OSM>SQLDelight>MapCompose doesn't unless I move the map away and back again.

Thanks in advance!
Jim

**Comment by p-lr** [comment, marked as answer]
If I understand correctly, when you have new tiles available in your database, you would like MapCompose to update.
Then, you can use `MapState.reloadTiles()` api.

**Reply by JakkeJ** [reply]
While I did not do this, I might try it later. I did however fork it and add an "isLoaded" boolean that I passed through, with a placeholder bitmap. Didn't think I would get it to work so I asked here, then I got it working shortly afterwards, like it usually is! :)

**Comment by Husoid** [comment]
Please share an example of how you implemented uploading to the database and uploading to the database with OSM in the absence of

**Reply by JakkeJ** [reply]
I can't make an example on my phone, going to bed, but I can tell you what I did.

So I had a repository for downloading tiles from OSM or whatever provider you want to use.

This repository first checked the local database, if there was a tile it would add it to a Buffer and emit it. If there was none, it would download the tile, add it to the buffer and emit it, but also add it to the local database so it was stored for next time you'd need to load that tile again.
